packages/sql/src/visit/visitors.js

```diff
@@ -0,0 +1,103 @@
+import { AGGREGATE, COLUMN_REF, FRAGMENT, PARAM, VERBATIM, WINDOW } from '../constants.js';
+import { aggregateNames, AggregateNode } from '../ast/aggregate.js';
+import { ColumnRefNode } from '../ast/column-ref.js';
+import { SQLNode } from '../ast/node.js';
+import { walk } from './walk.js';
+
+const aggrRegExp = new RegExp(`^(${aggregateNames.join('|')})`);
+const funcRegExp = /(\\'|\\"|"(?:\\"|[^"])*"|'(?:\\'|[^'])*'|\$\w+|\w+\()/g;
```

Comment:
Could be more readable with comments.

```suggestion
const funcRegExp = /(
    \\'               | // Matches an escaped single quote (e.g., \')
    \\"               | // Matches an escaped double quote (e.g., \")
    "(?:\\"|[^"])*"   | // Matches a double-quoted string, allowing for escaped quotes (e.g., "text with \"escaped\" quotes")
    '(?:\\'|[^'])*'   | // Matches a single-quoted string, allowing for escaped quotes (e.g., 'text with \'escaped\' quotes')
    \$\w+             | // Matches a variable starting with $ followed by word characters (e.g., $variableName)
    \w+\(               // Matches a function name, capturing any word characters followed by an opening parenthesis (e.g., functionName()
)/g;
```